lib/src/frontend/spawn_hybrid.dart

```diff
@@ -0,0 +1,166 @@
+/// isolate is killed, the channel's stream will emit a "done" event.
+///
+/// Any unhandled errors loading or running the hybrid isolate will be emitted
+/// as errors over the channel's stream. Any calls to `print()` in the hybrid
+/// isolate will be printed as though they came from the test that created the
+/// isolate.
+///
+/// Code in the hybrid isolate is not considered to be running in a test
+/// context, so it can't access test functions like `expect()` and
+/// `expectAsync()`.
+///
+/// **Note**: If you use this API, be sure to add a dependency on the
+/// **`stream_channel` package, since you're using its API as well!
+StreamChannel spawnHybridUri(uri, {Object message}) {
+  Uri parsedUrl;
+  if (uri is Uri) {
+    parsedUrl = uri;
+  } else if (uri is String) {
+    parsedUrl = Uri.parse(uri);
+  } else {
+    throw new ArgumentError.value(uri, "uri", "must be a Uri or a String.");
+  }
+
+  String uriString;
+  if (parsedUrl.scheme.isEmpty) {
+    var suitePath = Invoker.current.liveTest.suite.path;
+    uriString = p.url.join(
+        p.url.dirname(p.toUri(p.absolute(suitePath)).toString()),
+        parsedUrl.toString());
+  } else {
+    uriString = uri.toString();
+  }
+
+  return _spawn(uriString, message);
+}
+
+/// Spawns a VM isolate that runs the given [dartCode], which is loaded as the
+/// contents of a Dart library.
+///
+/// This allows browser tests to spawn servers with which they can communicate
+/// to test client/server interactions. It can also be used by VM tests to
+/// easily spawn an isolate.
+///
+/// The [dartCode] must define a top-level `hybridMain()` function that takes a
+/// `StreamChannel` argument and, optionally, an `Object` argument to which
+/// [message] will be passed. Note that [message] must be JSON-encodable. For
+/// example:
+///
+/// ```dart
+/// import "package:stream_channel/stream_channel.dart";
+///
+/// hybridMain(StreamChannel channel, Object message) {
+///   // ...
+/// }
+/// ```
+///
+/// Returns a [StreamChannel] that's connected to the channel passed to
+/// `hybridMain()`. Only JSON-encodable objects may be sent through this
+/// channel. If the channel is closed, the hybrid isolate is killed. If the
+/// isolate is killed, the channel's stream will emit a "done" event.
+///
+/// Any unhandled errors loading or running the hybrid isolate will be emitted
+/// as errors over the channel's stream. Any calls to `print()` in the hybrid
+/// isolate will be printed as though they came from the test that created the
+/// isolate.
+///
+/// Code in the hybrid isolate is not considered to be running in a test
+/// context, so it can't access test functions like `expect()` and
+/// `expectAsync()`.
+///
+/// **Note**: If you use this API, be sure to add a dependency on the
+/// **`stream_channel` package, since you're using its API as well!
+StreamChannel spawnHybridCode(String dartCode, {Object message}) {
```

Comment:
DBC: This seems cool – but feel scary.

Is there a strong case for this over just giving folks the URI-based method?